### PR TITLE
tests: lib: cmsis_nn: move to new ztest API

### PR DESCRIPTION
**Pull request by yerabolu**
Move lib tests to use new ztest API

Part of #47002 effort

Signed-off-by: Spoorthy Priya Yerabolu <spoorthy.priya.yerabolu@intel.com>

### CODE REVIEW

**Review by JordanYates — DISMISSED** [review]

File: tests/lib/cmsis_nn/src/main.c [hunk 1]
```diff
@@ -146,7 +146,7 @@ const q7_t conv_4_output_ref[36] = {
 	4, 5, 14, 2, 5, 7, -1, -2, 2, 5, -4, 11, -1, -2, 8, 4, 2, 0
 };
 
-void test_convolve(void)
+ZTEST(test_cmsis_nn, test_convolve)
```
Comment (nashif) [inline tests/lib/cmsis_nn/src/main.c:149]: remove leading test_ from testsuite name.

**Review by stephanosio — CHANGES_REQUESTED** [review]
https://github.com/zephyrproject-rtos/zephyr/pull/47109#discussion_r911518414

File: tests/lib/devicetree/api/prj.conf [hunk 2]
```diff
@@ -1 +1,2 @@
 CONFIG_ZTEST=y
+CONFIG_ZTEST_NEW_API=y
```
Comment (stephanosio) [inline tests/lib/devicetree/api/prj.conf:2]: ?

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: tests/lib/devicetree/api/prj.conf (hunk 2 above)
Comment (yerabolu) [reply tests/lib/devicetree/api/prj.conf:2]: Sorry, other test changes got into the commit by mistake.  Working on cleaning this.

**Review by stephanosio — APPROVED** [review]

File: tests/lib/cmsis_nn/src/main.c (hunk 1 above)
Comment (yerabolu) [reply tests/lib/cmsis_nn/src/main.c]: Removed test_ for testsuite name.

**Review by enjiamai — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**